### DISCUSSION TITLE
How to load large table in read_sql()?

### DISCUSSION THREAD

**Discussion by alexoff13** [atoti/atoti#616] [Q&A]
Hello!
I have a table in ClickHouse with 400 million rows and 45 columns
Based on it, I want to build a cube. When calling the `read_sql()` method, I can't wait for it to complete.
Tell me, is atoti generally able to work with such a volume of information?

**Comment by mailys-ds** [comment]
Hi,


First, would it be possible to know your dataset size? If I approximatly estimate it, it should be around 72GB.

Then two factors will be responsible for the `read_sql()` latency time:
- Your available memory. As atoti stores all data in memory, you need to have this amount of memory available.
- Your internet speed.

So considering you have enough memory available, loading data with a speed of 80Mbps, you might still need roughly 1 hour 40 min.


On my side, I tried to load the [clickhouse tutorial dataset](https://clickhouse.com/docs/en/tutorial) with 2 millions lines and 45 columns, it took a bit less than 30 seconds. 
![image](https://user-images.githubusercontent.com/62394323/174821885-0c68c00f-723d-4315-88f8-c172192c514b.png). 
Then, querying on this dataset was pretty straighforward.
So, I'm surprised `read_sql()` did not work with more than 100.000 lines of your dataset. Could I get more precision about your dataset (size, column types).

Thanks.
